examples/platform/linux/AppMain.cpp

```diff
@@ -281,7 +282,13 @@ int ChipLinuxAppInit(int argc, char * const argv[], OptionSet * customOptions)
     }
     else if (LinuxDeviceOptions::GetInstance().traceStreamToLogEnabled)
     {
-        gTraceStream = new chip::trace::TraceStreamLog();
```

Comment:
This changes the format of the tracing: it forces the tracing to go through the new decoder **and loses the machine-readable raw logs that were available before** this change.

Reply:
Could we just add a new command line argument for this style of trace output? 

Something like `--trace_decode` instead of replacing the `--trace_log`, that way it's just adding new functionality and a supporting tool, and not changing any of the existing behaviour.

Reply:
I have updated the PR such that `--trace_log 1` keeps the original behaviour for both `chip-tool` and the examples apps.
However `--trace_log 2` does the "extended (bespoke version)